android-components/components/service/nimbus/src/main/java/mozilla/components/service/nimbus/messaging/NimbusMessagingStorage.kt

```diff
@@ -260,23 +266,19 @@ class NimbusMessagingStorage(
     fun isMessageEligible(
         message: Message,
         helper: NimbusMessagingHelperInterface,
-        jexlCache: MutableMap<String, Boolean> = mutableMapOf(),
     ): Boolean {
         return message.triggers.all { condition ->
-            jexlCache[condition]
-                ?: try {
-                    if (malFormedMap.containsKey(condition)) {
-                        return false
-                    }
-                    helper.evalJexl(condition).also { result ->
-                        jexlCache[condition] = result
-                    }
-                } catch (e: NimbusException.EvaluationException) {
-                    reportMalformedMessage(message.id)
-                    malFormedMap[condition] = message.id
-                    logger.info("Unable to evaluate $condition")
-                    false
+            try {
+                if (malFormedMap.containsKey(condition)) {
+                    return false
                 }
+                helper.evalJexl(condition)
```

Comment:
Similar to this [comment](https://github.com/mozilla-mobile/firefox-android/pull/5652/files#r1494791061), could we call `use` here so we don't wrap the parent functions in use?